examples/provider_test.go

```diff
@@ -317,3 +324,32 @@ func planEqual(t *testing.T, firstPlan, secondPlan string) bool {
 
 	return assert.Equal(t, firstPlanData, secondPlanData)
 }
+
+func inPlacePulumiTest(t *testing.T, dir string, opts ...opttest.Option) *pulumitest.PulumiTest {
```

Comment:
This allows us now to use an in-process provider for the tests. Which will also give us code coverage :)

Reply:
This is good but might want to be careful to not use this for production release verification in case there's accidentally some behavioral difference between the prod binary and this form of running the provider.

Reply:
Yeah, before adopting this for all tests we should add some sort of flag/env variable to fall back to attaching the binary for releases